epan/dissectors/packet-ippusb.c

```diff
@@ -268,95 +283,109 @@ dissect_ippusb(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree, void* data)
 
                 if (previous_msp) {
                     previous_msp->nxtpdu = pinfo->num;
-                    new_msp = pdu_store(pinfo, ippusbd->multisegment_pdus, previous_msp->first_frame, previous_msp->is_chunked, previous_msp->http_ipp_split);
-                    new_msp->running_size = previous_msp->running_size + reported_length;
-
-                    if(last != TAG_END_OF_ATTRIBUTES) {
-                        if(reported_length <= CHUNK_LENGTH_MAX){
-                            /* This segment has the size of the next packet as a part
-                             * of the  HTTP 1.1 chunked tranfer encoding */
-
-                            new_msp->running_size -= reported_length;
-                            new_msp->is_chunked = TRUE;
-
-                            if(new_msp->http_ipp_split == HTTP_IPP_SPLIT_NOT_SET){
-                                new_msp->running_size += sizeof(CHUNKED_END);
-                                new_msp->http_ipp_split = new_msp->running_size;
-
-                                fragment_add_check(&ippusb_reassembly_table, chunked_end_tvb, offset, pinfo, new_msp->first_frame,
-                                            GUINT_TO_POINTER(new_msp->first_frame), previous_msp->running_size, sizeof(CHUNKED_END), TRUE);
-                            }
-                            else {
-                                return reported_length;
-                            }
+                    new_msp = pdu_store(pinfo, ippusbd->multisegment_pdus, previous_msp->first_frame, previous_msp->is_ipp, previous_msp->document);
+                    new_msp->running_size = previous_msp->running_size + captured_length;
+
+                    /* THis packet has an HTTP header but is not an ipp packet */
+                    if ((first_linelen >= 14 && strncmp(first_line, "Content-Type: ", 14) == 0) &&
+                        (first_linelen < 29 || strncmp(first_line, "Content-Type: application/ipp", 29) != 0)) {
+                        new_msp->is_ipp = FALSE;
+                    }
+
+                    /* This packet will have an attached document */
+                    if (status_code == PRINT_JOB || status_code == SEND_DOCUMENT) {
+                        new_msp->document |= MSP_HAS_DOCUMENT;
+                    }
+
+                    if(!(last_chunk && strncmp(last_chunk, CHUNKED_END, CHUNK_LENGTH_MIN) == 0)){
```

Comment:
>if(!(last_chunk && strncmp(last_chunk, CHUNKED_END, CHUNK_LENGTH_MIN) == 0)){ [](start = 20, length = 77)

It's just difficult to mentally parse the logic - a code comment below (something like' if this is not the last chunk in the chunked transfer' goes a long way)